Assets/Spriter2UnityDX/Editor/AnimationBuilder.cs

```diff
@@ -83,6 +86,36 @@ public class AnimationBuilder : UnityEngine.Object {
 					}
 				}
 			}
+            if (animation.soundlines == null)
+            {
+                //this means there are no sounds
+                //set events to null in case there were sounds on a previous import
+                AnimationUtility.SetAnimationEvents(clip, null);
+            }
+            else
+            {
+                HasSounds = true;
+                List<AnimationEvent> soundEvents = new List<AnimationEvent>();
+
+                foreach (SoundLine sLine in animation.soundlines)
+                {
+                    foreach (TimeLineKey key in sLine.keys)
+                    {
+                        AnimationEvent sEvent = new AnimationEvent();
+                        sEvent.functionName = "playSoundEffect";
+
+                        SpriteInfo sInfo = (SpriteInfo)key.info;
+                        sEvent.objectReferenceParameter = SoundFolders[sInfo.folder][sInfo.file];
+
+                        sEvent.time = key.time;
+                        soundEvents.Add(sEvent);
+                    }
+                }
+                //events must be in choronological order or they won't fire properly
+                soundEvents.Sort((a, b) => a.time.CompareTo(b.time));
+                AnimationUtility.SetAnimationEvents(clip, soundEvents.ToArray());
```

Comment:
As a bit of a suggestion, take a look at how my TextureController interacts with the AnimationClip. It changes the texture in response to a value being changed by the animation, instead of using AnimationEvents.